subprojects/internal-performance-testing/src/main/groovy/org/gradle/performance/fixture/BuildExperimentRunner.java

```diff
@@ -85,6 +87,18 @@ public void run(BuildExperimentSpec experiment, MeasuredOperationList results) {
         }
     }
 
+    private void configureCompilerDaemonAndTestWorkerMemory(BuildExperimentSpec experiment, File workingDirectory) {
```

Comment:
I wouldn't modify the build file, but put the configuration into the example project directly, so that it can be overwritten with a `-P` property:

```
def compilerMemory = findProperty('compilerMemory') ?: '512m'
tasks.withType(JavaCompile) {
  options.forkOptions.jvmArgs = ["-Xmx$compilerMemory"]
}
```

Then in the test:

```
runner.args = ["-PcompilerMemory=128m"]
```

That way it is easy to use when testing with the Gradle Profiler or testing manually.